ci/functional/test_main_prep_node.sh

```diff
@@ -2,33 +2,220 @@
 
 set -eux
 
-for i in 0 1; do
+: "${FIRST_NODE:=}"
+: "${OPERATIONS_EMAIL:=}"
+
+result=0
+mail_message=''
+nl="
+"
+
+function do_mail {
+    if [ -z "$mail_message" ]; then
+        return
+    fi
+    set +x
+    if [ -z "$OPERATIONS_EMAIL" ]; then
+        echo "$mail_message"
+        return
+    fi
+    # shellcheck disable=SC2059
+    mail -s "Hardware check failed after reboot!" \
+         -r "$HOSTNAME"@intel.com "$OPERATIONS_EMAIL" \
+         <<< "$mail_message"
+    set -x
+}
+
+if ! command -v lspci; then
+    if command -v dnf; then
+       dnf -y install pciutils
+    else
+       echo "pciutils not installed, can not test for Infiniband devices"
+    fi
+fi
+
+hdr_count=0
+opa_count=0
+ib_count=-1
+
+set +x
+while IFS= read -r line; do
+    ((opa_count++)) || true
+done < <(lspci --mm | grep "Omni-Path")
+if [ $opa_count -gt 0 ]; then
+    ((ib_count=opa_count)) || true
+fi
+
+while IFS= read -r line; do
+    mlnx_type="${line##*-}"
+    mlnx_type="${mlnx_type%*]}"
+    if [ "$mlnx_type" -ge 6 ]; then
+        ((hdr_count++)) || true
+    fi
+done < <(lspci -mm | grep "ConnectX")
+if [ $hdr_count -gt 0 ]; then
+    ((ib_count=hdr_count)) || true
+fi
+
+# Can not have Omni-Path and Mellanox HDR on the same system.
+if [ $hdr_count -gt 0 ] && [ $opa_count -gt 0 ]; then
+    ib_message="Invalid hardware configuration.  Found:
+$hdr_count Mellanox HDR ConnectX adapters,
+and
+$opa_count Omni-Path adapters.
+The Onmi-Path adapters will not be used."
+    mail_message+="${nl}${ib_message}${nl}"
+    echo "$ib_message"
+fi
+set -x
+
+function do_wait_for_ib {
+    ib_timeout=300 # 5 minutes
+    retry_wait=10 # seconds
+    timeout=$((SECONDS + ib_timeout))
+    while [ "$SECONDS" -lt "$timeout" ]; do
+      ip link set up "$1" || true
+      sleep 2
+      if ip addr show "$1" | grep "inet "; then
+        return 0
+      fi
+      sleep ${retry_wait}
+    done
+    return 1
+}
+
+# First check for infinband devices
+for i in $(seq 0 $((ib_count-1))); do
```

Comment:
(lint) Couldn't parse this for loop. Fix to allow more checks. [SC1073]
(lint) Couldn't find 'done' for this 'do'. [SC1061]